src/utils.ts

```diff
@@ -19,13 +21,91 @@ export function WebWorker(worker: Worker) {
  * Fallback for localStorage that works in node and the browser
  */
 export function getPlatformLocalStorage() {
-  if (typeof window === "undefined" || window === null) {
-    // eslint-disable-next-line @typescript-eslint/no-var-requires
+  if (isNodeEnv()) {
     const LocalStorage = require("node-localstorage").LocalStorage;
     return new LocalStorage("./local-storage");
-  } else {
+  }
+  if (isWebEnv()) {
     return window.localStorage;
   }
+  throw new Error(
+    "Please pass in valid localStorage object into the Hedgehog constructor"
+  );
+}
+
+function isReactNativeEnv() {
+  return (
+    typeof navigator !== "undefined" && navigator.product === "ReactNative"
+  );
+}
+
+function isWebEnv() {
+  return (
+    typeof window !== "undefined" &&
+    window &&
+    window.Worker &&
+    !isReactNativeEnv()
+  );
+}
+
+function isNodeEnv() {
+  return typeof window === "undefined" || window === null;
+}
+
+export function getPlatformCreateKey() {
+  /**
+   * Given a user encryptStr and initialization vector, generate a private key
+   * @param encryptStr String to encrypt (can be user password or some kind of lookup key)
+   * @param ivHex hex string iv value
+   */
+  const createKey = async (encryptStr: string, ivHex: string) => {
+    return new Promise<PrivateKey>((resolve, reject) => {
+      if (isWebEnv()) {
```

Comment:
I guess using the `getPlatformCreateKey` wrapper isn't really necessary bc the branching happens inside `createKey`. We could probably use `createKey` as the default arg to hedgehog

But also some value in being consistent with `getPlatformLocalStorage`. wdyt?

Reply:
yeah this is a great call, i was ideally following the pattern with local-storage, and it's just a coincidence that the branching happens in a promise vs having two promises. maybe i will divide the promises up actually?